File: src/main/java/me/yoursole/ctf/Events/MoveItemEvent.java
package me.yoursole.ctf.Events;

import me.yoursole.ctf.DataFiles.Items.Flag;
import org.bukkit.ChatColor;
import org.bukkit.entity.Player;
import org.bukkit.event.EventHandler;
import org.bukkit.event.Listener;
import org.bukkit.event.inventory.InventoryClickEvent;
import org.bukkit.event.inventory.InventoryCloseEvent;
import org.bukkit.event.inventory.InventoryOpenEvent;
import org.bukkit.inventory.Inventory;

import java.util.ArrayList;

public class MoveItemEvent implements Listener {
    private static ArrayList<Player> openers = new ArrayList<>();
    @EventHandler
    public void onMoveItem(InventoryClickEvent e){
        if(e.getCurrentItem()!=null&&openers.contains(e.getWhoClicked())&&e.getCurrentItem().isSimilar(Flag.flag)){
            e.setCancelled(true);
            e.getWhoClicked().sendMessage(ChatColor.RED+"You can not get rid of this");
        }
    }

    @EventHandler
    public void onOpenGUI(InventoryOpenEvent e){
        if(e.getInventory()!=e.getPlayer().getInventory())
            openers.add((Player) e.getPlayer());
    }

    @EventHandler
    public void onCloseGUI(InventoryCloseEvent e){
        openers.remove((Player) e.getPlayer());
    }

}

File: src/main/kotlin/me/yoursole/ctf/events/MoveItemEvent.kt
package me.yoursole.ctf.events

import me.yoursole.ctf.datafiles.items.Flag
import org.bukkit.entity.Player
import org.bukkit.event.EventHandler
import org.bukkit.event.Listener
import org.bukkit.event.inventory.InventoryClickEvent
import org.bukkit.event.inventory.InventoryCloseEvent
import org.bukkit.event.inventory.InventoryOpenEvent

object MoveItemEvent : Listener {
    @EventHandler
    fun onMoveItem(e: InventoryClickEvent) {
        if (e.currentItem != null && openers.contains(e.whoClicked) && e.currentItem!!
                .isSimilar(Flag.flag)
        ) {
            e.isCancelled = true
            e.whoClicked.sendMessage("§cYou can not get rid of this")
        }
    }

    @EventHandler
    fun onOpenGUI(e: InventoryOpenEvent) {
        if (e.inventory !== e.player.inventory) openers.add(e.player as Player)
    }

    @EventHandler
    fun onCloseGUI(e: InventoryCloseEvent) {
        openers.remove(e.player as Player)
    }

    private val openers = arrayListOf<Player>()
}